### PR TITLE
UI overhaul, fix crash/reload

### PR DESCRIPTION
**Pull request by easyaspi314**
Major redesign in preferences!

* Fixed the reload on orientation and NullPointerException crash
* We still need to deal with App Info crashes on 4.1. I haven't implemented anything yet because I am not familiar with Xposed.

**\*\*Everything is English only and will need to be translated eventually****
* Help screen on ActionBar - Should appease for now, but should eventually be updated with locale-specific and API-specific things, and load from online to keep the module small.
 * Dialog on tablet, full-screen on phone
 * Also fixed the dead link.
* Added basic summaries on some of the confusing App Info preferences.
* Preference Headers for dual-column in tablets. Also preferences are separated into Fragments.
* Cleaned up the code a bit. Moved OnPreferenceClickListeners to being implemented in the class so it is less messy and used string-switching (note: You may need to configure Eclipse to keep it from yelling at you. IDK, I used [AIDE](https://play.google.com/store/apps/details?id=com.aide.ui) on my phone.
* Reorganized app-specific stuff
* Detects if the module is disabled by hooking into itself.
 * Shows a message at the bottom of the screen.
* Removed the silly empty PreferenceCategory at the bottom of the App Info screen.

### CODE REVIEW

File: src/com/pyler/xinstaller/Preferences.java [hunk 1]
```diff
@@ -39,6 +39,7 @@ protected void onCreate(Bundle savedInstanceState) {
 		@Override
 		public void onCreate(Bundle savedInstanceState) {
 			super.onCreate(savedInstanceState);
+            context = getActivity();
```
Comment (pylerSM) [inline src/com/pyler/xinstaller/Preferences.java]: add space please

**Comment by pylerSM** [comment]
Good good :)

File: src/com/pyler/xinstaller/Preferences.java (hunk 1 above)
Comment (easyaspi314) [inline src/com/pyler/xinstaller/Preferences.java]: Sorry I use AIDE and it doesn't learn formatting. Will fix.

Comment (easyaspi314) [inline src/com/pyler/xinstaller/Preferences.java]: Right now I am cleaning up the preferences. So far I moved the preference
keys to Common and I am working on moving to preference headers.

Comment (pylerSM) [inline src/com/pyler/xinstaller/Preferences.java]: Great :)

**Comment by easyaspi314** [comment]
Hey, I have a question: Should we use Holo/Material themes or stay with DeviceDefault? (Because preference headers are ugly in TW).

**Comment by easyaspi314** [comment]
Here are some screenshots of my current thing using App Settings and Holo Themer to switch themes and Tablet Layout.

DeviceDefault
screenshot_2015-10-20-07-10-17 
screenshot_2015-10-20-07-10-22
screenshot_2015-10-20-07-08-22

Holo
screenshot_2015-10-20-07-09-53
screenshot_2015-10-20-07-09-56
screenshot_2015-10-20-07-07-01

Holo Light/Dark
screenshot_2015-10-20-07-09-22
screenshot_2015-10-20-07-09-29
screenshot_2015-10-20-07-08-49

**Comment by pylerSM** [comment]
I prefer DeviceDefault theme :)

**Comment by pylerSM** [comment]
Maybe you can add item "Bug report" which opens browser with link of issue page of this Github project.

**Comment by easyaspi314** [comment]
And source code. Good idea!

**Comment by easyaspi314** [comment]
I think I am on to something, but it seems that I am having [some fragment trouble.](https://drive.google.com/file/d/0Bxk_L8QgNv7TTHlyWW9ra0hRZHc/view?usp=sharing) What can I say. Fragments hate me. :)

**Comment by pylerSM** [comment]
heh ;)

**Comment by easyaspi314** [comment]
I am likely going to commit later today. I just got a new SD card (8GB Class 2 -> 32GB UHS-1 :smile: ) and am currently moving stuff over... but Vista copies slooooowly.

**Comment by easyaspi314** [comment]
I would've committed earlier but I had issues with the things in the video, `removePreference(Preference preference`, and SwitchPreference's OnClickListener bug. I am currently implementing a module disabled notification and then it might just be ready. Still need to fix 4.1 issues though but I don't know Xposed well.

**Comment by pylerSM** [comment]
I will fix that 4.1 issue myself. That's okay.

**Comment by easyaspi314** [comment]
OK I am almost done, I am just having a bit of trouble with recreate() and
Preference Headers. I will try and finish it up this afternoon.

– Devin Hussey

**Comment by easyaspi314** [comment]
Here ya go. Test it and see if it works on your device(s) and can you do me
a favor and test it on a tablet (or tablet-sized emulator)?

**Comment by easyaspi314** [comment]
Feel free to reformat it, I may have used tabs instead of spaces and I
think a few extra lines.

**Comment by easyaspi314** [comment]
By the way to open apps in tablet mode I used App Settings. Just lower the DPI to something like 160 and set the Screen (dp) to 1000x1600. Some things might not work properly because unused themes are usually removed from your ROM to save space.

**Comment by easyaspi314** [comment]
You have write access.

**Comment by pylerSM** [comment]
Crash: Click on report bug, cancel dialog, click on Github - crash.

Something with resuming and preferences.

**Comment by easyaspi314** [comment]
Can you show a logcat? (auto correct changed to logical (: )

– Devin Hussey

**Comment by easyaspi314** [comment]
nvm try what I have now.

– Devin Hussey

**Comment by pylerSM** [comment]
http://pastebin.com/p8LQdHj1

**Comment by pylerSM** [comment]
11-05 18:26:44.261 E/AndroidRuntime( 7603): FATAL EXCEPTION: main
11-05 18:26:44.261 E/AndroidRuntime( 7603): java.lang.RuntimeException: Unable to resume activity {com.pyler.xinstaller/com.pyler.xinstaller.Preferences}: java.lang.NullPointerException
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2830)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2859)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1322)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.os.Handler.dispatchMessage(Handler.java:99)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.os.Looper.loop(Looper.java:137)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.ActivityThread.main(ActivityThread.java:5283)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at java.lang.reflect.Method.invokeNative(Native Method)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at java.lang.reflect.Method.invoke(Method.java:511)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1102)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:869)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at dalvik.system.NativeStart.main(Native Method)
11-05 18:26:44.261 E/AndroidRuntime( 7603): Caused by: java.lang.NullPointerException
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.preference.PreferenceGroup.removePreferenceInt(PreferenceGroup.java:188)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.preference.PreferenceGroup.removePreference(PreferenceGroup.java:181)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at com.pyler.xinstaller.Preferences$Settings.onResume(Preferences.java:339)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.Fragment.performResume(Fragment.java:1742)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:937)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1075)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1057)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.FragmentManagerImpl.dispatchResume(FragmentManager.java:1870)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.Activity.performResume(Activity.java:5371)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2820)
11-05 18:26:44.261 E/AndroidRuntime( 7603): 	... 11 more

**Comment by easyaspi314** [comment]
OK. I think I see the issue. I soft-bricked my phone so I can't test it. :angry: 

Try moving all of the onResume() code to onCreate() in the fragment.
See if it happens again. It seems that appTranslator is null because it has already been removed.

**Comment by easyaspi314** [comment]
I was planning to deregister the listeners in onPause(), but that never happened.

**Comment by pylerSM** [comment]
Another reported crash by user on XDA: https://www.dropbox.com/s/m6v9l3d2uwr5whc/alogcat.2015-11-11-11-55-40-0200.txt?dl=0

**Comment by pylerSM** [comment]
And it seems we have another issue too: https://github.com/pylerSM/XInstaller/issues/28

Maybe I will have to move back to old UI, since we need working UI for work on XInstaller for Android M.

**Comment by pylerSM** [comment]
Okey, I saw and merged your fixes. We need to verify #28 .

**Comment by easyaspi314** [comment]
I got LP running and your module causes my phone to just hang at boot. It definitely looks like the preferences are the issue. Maybe as a hack, we could do something like this:

com.pyler.xinstaller.Preferences
```java
@Override
public void onPause() {
    super.onPause();
    try {
        new XSharedPreferences("com.pyler.xinstaller").makeWorldReadable();
    } catch (Throwable e) {}
}
```

**Comment by pylerSM** [comment]
this fix will not fix it

getPreferenceManager().setSharedPreferencesMode(
					Context.MODE_WORLD_READABLE); 

should handle it.

when we edit something manually we need to open it as https://github.com/pylerSM/XInstaller/blob/master/src/com/pyler/xinstaller/Utils.java#L220


This bug is release-blocker. Need good fix, not fast hack.

**Comment by pylerSM** [comment]
The problem is that after you change any preference, prefs file became not "world readable". I don't know why since we open it in this mode.

**Comment by pylerSM** [comment]
This (https://github.com/pylerSM/XInstaller/blob/e9ac602542159ccfa77c907e67c5dae93e35368a/src/com/pyler/xinstaller/Preferences.java#L104) is bad code, imho, since we should open it in world readable mode.

**Comment by pylerSM** [comment]
@easyaspi314 We need to fix issue https://github.com/pylerSM/XInstaller/pull/27#issuecomment-154132019 and this is ready to go as stable release.

**Pull request merged**